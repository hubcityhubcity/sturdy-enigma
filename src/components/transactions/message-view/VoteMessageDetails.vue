<template>
  <div>
    <div class="tx__content__caption">
      <p>Voted&nbsp;{{ transaction.value.option }}</p>
    </div>
    <div class="tx__content__information">
      On&nbsp;
      <router-link :to="`/governance/${transaction.value.proposal_id}`"
        >Proposal &#35;{{ transaction.value.proposal_id }}</router-link
      >
    </div>
  </div>
</template>

<script>
import { formatBech32 } from "src/filters"

export default {
  name: `vote-message-details`,
  filters: {
    formatBech32
  },
  props: {
    transaction: {
      type: Object,
      required: true
    },
    validators: {
      type: Object,
      required: true
    }
  }
}
</script>
