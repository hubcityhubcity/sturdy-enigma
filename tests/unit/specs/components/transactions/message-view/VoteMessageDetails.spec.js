import { shallowMount } from "@vue/test-utils"
import VoteMessageDetails from "src/components/transactions/message-view/VoteMessageDetails"

describe(`VoteMessageDetails`, () => {
  let wrapper
  // Examples @ https://stargate.cosmos.network/txs?action=vote
  const tx = {
    height: 2000000,
    value: {
      proposal_id: "5",
      voter: "cosmos1lktjhnzkpkz3ehrg8psvmwhafg56kfss5597tg",
      option: "Abstain"
    },
    fee: {
      amount: "37",
      denom: "uatom"
    },
    hash: `ABCD1234`,
    group: `governance`,
    key: "keyhash",
    memo: "(Sent via Lunie)",
    timestamp: 123456789,
    liquidDate: null,
    type: `cosmos-sdk/MsgVote`
  }

  it(`renders a vote transaction message`, () => {
    wrapper = shallowMount(VoteMessageDetails, {
      propsData: {
        transaction: tx,
        validators: {}
      },
      stubs: [`router-link`]
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
