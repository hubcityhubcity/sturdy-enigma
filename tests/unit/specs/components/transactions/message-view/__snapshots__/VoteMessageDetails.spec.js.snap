// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoteMessageDetails renders a vote transaction message 1`] = `
<div
  class="tx__content"
>
  <transactionicon-stub
    transactiongroup="governance"
    transactiontype="Voted"
  />
   
  <div
    class="tx__content__left"
  >
    <h3>
      Voted
    </h3>
     
    <span>
      <i
        class="italic"
      >
        Abstain
      </i>
       onÂ 
    </span>
     
    <router-link-stub
      to="/governance/5"
    >
      Proposal #5
    </router-link-stub>
  </div>
</div>
`;
